From: Juergen Beisert <jbe@pengutronix.de>
Subject: Mini2440: Enable buzzer

Signed-off-by: Juergen Beisert <jbe@pengutronix.de>
---
 arch/arm/mach-s3c2440/mach-mini2440.c |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

Index: linux-3.2/arch/arm/mach-s3c2440/mach-mini2440.c
===================================================================
--- linux-3.2.orig/arch/arm/mach-s3c2440/mach-mini2440.c
+++ linux-3.2/arch/arm/mach-s3c2440/mach-mini2440.c
@@ -606,6 +606,15 @@ static struct s3c_hwmon_pdata mini2440_a
 	.in[7] = &mini2440_adc_chcfg[7],
 };
 
+static struct platform_device mini2440_buzzer_device = {
+	.name = "pwm-beeper",
+	.id = -1,
+	.dev = {
+		.parent = &s3c_device_timer[0].dev,
+		.platform_data = (void *)0,	/* channel 0 */
+	},
+};
+
 static struct platform_device *mini2440_devices[] __initdata = {
 	&s3c_device_ohci,
 	&s3c_device_wdt,
@@ -623,6 +632,8 @@ static struct platform_device *mini2440_
 	&samsung_asoc_dma,
 	&s3c_device_adc,
 	&s3c_device_hwmon,
+	&s3c_device_timer[0],
+	&mini2440_buzzer_device,
 };
 
 static void __init mini2440_map_io(void)
@@ -786,6 +797,9 @@ static void __init mini2440_init(void)
 	i2c_register_board_info(0, mini2440_i2c_devs,
 				ARRAY_SIZE(mini2440_i2c_devs));
 
+	/* PWM to the buzzer */
+	s3c_gpio_cfgpin(S3C2410_GPB(0), S3C2410_GPB0_TOUT0);
+
 	platform_add_devices(mini2440_devices, ARRAY_SIZE(mini2440_devices));
 
 	if (features.count)	/* the optional features */
