Author: Juergen Beisert <jbe@pengutronix.de>
Subject: Add the known phyical display sizes the Mini2440 uses

Signed-off-by: Juergen Beisert <jbe@pengutronix.de>

---
 arch/arm/mach-s3c2440/mach-mini2440.c |   35 +++++++++++++++++++++-------------
 1 file changed, 22 insertions(+), 13 deletions(-)

Index: linux-3.2/arch/arm/mach-s3c2440/mach-mini2440.c
===================================================================
--- linux-3.2.orig/arch/arm/mach-s3c2440/mach-mini2440.c
+++ linux-3.2/arch/arm/mach-s3c2440/mach-mini2440.c
@@ -109,7 +109,7 @@ static struct s3c2410_udc_mach_info mini
  * This macro simplifies the table bellow
  */
 #define _LCD_DECLARE(_clock,_xres,margin_left,margin_right,hsync, \
-			_yres,margin_top,margin_bottom,vsync, refresh) \
+			_yres,margin_top,margin_bottom,vsync, refresh, vwidth, vheight) \
 	.width = _xres, \
 	.xres = _xres, \
 	.height = _yres, \
@@ -120,6 +120,8 @@ static struct s3c2410_udc_mach_info mini
 	.lower_margin	= margin_bottom,	\
 	.hsync_len	= hsync,	\
 	.vsync_len	= vsync,	\
+	.phys_width	= vwidth,	\
+	.phys_height	= vheight,	\
 	.pixclock	= ((_clock*100000000000LL) /	\
 			   ((refresh) * \
 			   (hsync + margin_left + _xres + margin_right) * \
@@ -129,24 +131,26 @@ static struct s3c2410_udc_mach_info mini
 			   S3C2410_LCDCON1_TFT)
 
 static struct s3c2410fb_display mini2440_lcd_cfg[] __initdata = {
-	[0] = {	/* mini2440 + 3.5" TFT + touchscreen */
+	[0] = {	/* mini2440 + 3.5" TFT + touchscreen (NEC NL2432HC22-23B: N35) */
 		_LCD_DECLARE(
 			7,			/* The 3.5 is quite fast */
 			240, 21, 38, 6, 	/* x timing */
 			320, 4, 4, 2,		/* y timing */
-			60),			/* refresh rate */
+			60,			/* refresh rate */
+			55, 71),		/* physical size */
 		.lcdcon5	= (S3C2410_LCDCON5_FRM565 |
 				   S3C2410_LCDCON5_INVVLINE |
 				   S3C2410_LCDCON5_INVVFRAME |
 				   S3C2410_LCDCON5_INVVDEN |
 				   S3C2410_LCDCON5_PWREN),
 	},
-	[1] = { /* mini2440 + 7" TFT + touchscreen */
+	[1] = { /* mini2440 + 7" TFT + touchscreen (Innolux AT070TN83: N43/LCD70) */
 		_LCD_DECLARE(
 			10,			/* the 7" runs slower */
 			800, 40, 40, 48, 	/* x timing */
 			480, 29, 3, 3,		/* y timing */
-			50),			/* refresh rate */
+			50,			/* refresh rate */
+			153, 92),		/* physical size */
 		.lcdcon5	= (S3C2410_LCDCON5_FRM565 |
 				   S3C2410_LCDCON5_INVVLINE |
 				   S3C2410_LCDCON5_INVVFRAME |
@@ -163,17 +167,19 @@ static struct s3c2410fb_display mini2440
 			10,
 			1024, 1, 2, 2,		/* y timing */
 			768, 200, 16, 16, 	/* x timing */
-			24),	/* refresh rate, maximum stable,
+			24,	/* refresh rate, maximum stable,
 				 tested with the FPGA shield */
+			0, 0),			/* Size unknown */
 		.lcdcon5	= (S3C2410_LCDCON5_FRM565 |
 				   S3C2410_LCDCON5_HWSWP),
 	},
-	[3] = {	/* mini2440 + 3.5" TFT + TS -- New model as Nov 2009 "T35" */
+	[3] = {	/* mini2440 + 3.5" TFT + TS -- New model as Nov 2009 -- TD035STED4: T35 */
 		_LCD_DECLARE(
 			7,			/* The 3.5 is quite fast */
 			240, 21, 25, 6,		/* x timing */
 			320, 2, 4, 2,		/* y timing */
-			40),			/* refresh rate */
+			40,			/* refresh rate */
+			53, 71),		/* pyhsical size */
 		.lcdcon5	= (S3C2410_LCDCON5_FRM565 |
 				   S3C2410_LCDCON5_INVVLINE |
 				   S3C2410_LCDCON5_INVVFRAME |
@@ -186,18 +192,20 @@ static struct s3c2410fb_display mini2440
 			10,			/* the 5.3" runs slower */
 			640, 41, 68, 22,	/* x timing */
 			480, 26, 6, 2,		/* y timing */
-			40),			/* refresh rate */
+			40,			/* refresh rate */
+			113, 85),		/* pyhsical size */
 		.lcdcon5	= (S3C2410_LCDCON5_FRM565 |
 				   S3C2410_LCDCON5_INVVLINE |
 				   S3C2410_LCDCON5_INVVFRAME |
 				   S3C2410_LCDCON5_PWREN),
 	},
-	[5] = { /* mini2440 + 3,5" TFT + touchscreen -- SONY X35 */
+	[5] = { /* mini2440 + 3,5" TFT + touchscreen -- SONY ACX502BMU: X35 */
 		_LCD_DECLARE(
 			7,
 			240, 1, 26, 5,		/* x timing */
 			320, 1, 5, 9,		/* y timing */
-			60),			/* refresh rate */
+			60,			/* refresh rate */
+			55, 71),		/* pyhsical size */
 		.lcdcon5	= (S3C2410_LCDCON5_FRM565 |
 					S3C2410_LCDCON5_INVVDEN |
 					S3C2410_LCDCON5_INVVFRAME |
@@ -205,12 +213,13 @@ static struct s3c2410fb_display mini2440
 					S3C2410_LCDCON5_INVVCLK |
 					S3C2410_LCDCON5_HWSWP),
 	},
-	[6] = { /* LCD-W35i 3.5" display (LQ035Q1DG06)*/
+	[6] = { /* LCD-W35i 3.5" display (Sharp LQ035Q1DG06: W35i )*/
 		_LCD_DECLARE(
 			7,		/* clock */
 			320, 68, 66, 4, /* xres, margin_right, margin_left, hsync */
 			240, 4, 4, 9,	/* yres, margin_top, margin_bottom, vsync */
-			60),		/* refresh rate */
+			60,		/* refresh rate */
+			72, 54),	/* pyhsical size */
 		.lcdcon5	= (S3C2410_LCDCON5_FRM565 |
 					S3C2410_LCDCON5_INVVDEN |
 					S3C2410_LCDCON5_INVVFRAME |
